lib/API.js

```diff
@@ -851,9 +851,9 @@ API.prototype._startJson = function(file, opts, action, pipe, cb) {
   appConf.forEach(function(app) {
     if (opts.only && opts.only != app.name)
       return false;
-    if (opts.watch && opts.watch === true)
+    if (opts.watch && opts.watch === true && !app.watch)
       app.watch = true;
-    if (opts.ignore_watch)
+    if (opts.ignore_watch && !app.ignore_watch)
```

Comment:
can you reverse the check and test `!app.watch` before `opts` tests? Will be faster in that case ;). Same for the second condition `(!app.ignore_watch && opts.ignore_watch)`.

Maybe we can add a test [here](https://github.com/Unitech/pm2/blob/master/test/bash/pm2-dev.sh) to prevent regression?